cookbook/readme.md

```diff
@@ -0,0 +1,155 @@
+
+## In progress
+* [direct-database-actions](plugins/direct-database-actions)
+
+## Not started
+* [common-camel-patterns](plugins/common-camel-patterns)
+* [notifications-and-emails](plugins/notifications-and-emails)
+* [slimsgate-xml-advanced-patterns](plugins/slimsgate-xml-advanced-patterns)
+* [ui-only-actions-as-code](plugins/ui-only-actions-as-code)
+
+# How to use this cookbook
+
+**This section assumes you have already followed the instructions in the [top-level readme](../README.md) to setup your gradle.properties
+and run the `./gradlew idea` or `./gradlew eclipse` command successfully**
+
+**For extremely basic examples that can be used as jumping off points for new plugins, please see [templates](../templates)
+and the [top-level readme](../README.md) in this repo**
+
+This cookbook is intended to provide you with a series of example plugins and code samples for developing SLIMS plugins 
+utilizing the slimsgate api. It is not intended to replace the slims development manual, [Apache's camel documentation](https://camel.apache.org/components/4.0.x), 
+or [Thymeleaf's documentation](https://www.thymeleaf.org/doc/tutorials/3.1/usingthymeleaf.html). Rather, this is a 
+supplementary set of salient examples to demonstrate how these tools can join together and interact when creating your own Slimsgate plugins.
+
+## SLIMSGate API Services
+
+In particular, we want to highlight how you can utilize the services provided in the `PublicApi` class. This class can be found
+in your External Libraries in a directory like `/slimsservice-api-A.B.C/com/genohm/slims/server/service/PublicApi.class`. 
+
+The `A` `B` and `C` in `slimsservice-api-A.B.C` will be a number like `6.9.0`. You might have more than one of these folders:
+one is created for each version of SLIMS you have amongst your plugins' `apiVersion` values. For example, if you run `./gradlew idea` or `./gradlew eclipse`
+after creating a plugin that has the following `plugin.properties` file:
+
+```
+config-class=com.genohm.slims.custom.CustomConfiguration
+type=SLIMSGATE
+apiVersion=6.8.0
+```
+
+An external Library `slimsservice-api-6.8.0.jar` will be downloaded, and the contained version of `PublicApi.java` will
+be the one that the plugin utilizes. If you later change the version to 6.9.1 and run `./gradlew idea`, you will download a new 
+external library: `slimsservice-api-6.9.1.jar`.
+
+
+## What's inside?
+
+Under the [cookbook/plugins](plugins) folder you will find several sub-folders, broken into categories:
+
+<details>
+
+<summary>
+
+#### [common-camel-patterns](plugins/common-camel-patterns)
+
+</summary>
+
+   * Examples to illustrate commonly-utilized camel route designs and camel route endpoints other than "direct" routes that run some beans
+   * Concepts explored:
+     * [Netty](https://camel.apache.org/components/4.0.x/netty-component.html) (TCP connection plumbing, handled by camel)
```

Comment:
here same as previous comment -> put the 3.x version of the doc (https://camel.apache.org/components/3.14.x/index.html)